### DISCUSSION TITLE
#2 Second Sprint SRS, Angular, Benchmark Results

### DISCUSSION THREAD

**Discussion by Annalytic-programming** [CUMGroup/AMOGUS#7] [Blog]
Last week we were dealing with our SRS. In that we defined the scope and the use cases of the project as well as all the specific requirements. We won't go further into detail here as you can read the SRS if you are interested.
You can find the SRS to our project [here](https://github.com/CUMGroup/AMOGUS/blob/main/documents/requirements/SRS.md).

Further more we have (almost(?)) all mockups for our GUI and debated about the implementation of the functionality. 
We figured out that it is better to use Angular as our frontend. Two of our members already have experience with Angular, the other three will be doing a tutorial in the next Sprint.

We now have finished the Benchmarks for our analysis interpteter. Our Result:
- We decided to use a libary because of better runtime in checking expressions equality.
More about the tests you could find [here](https://github.com/CUMGroup/ExpressionInterpreterTest).

We hope that we could give you a quick insight into our second sprint and activities.

Yours truly

CUMGroup

**Comment by vans1646** [comment]
Hi I took a look at your SRS and the only point that you could maybe think about for the future is to write less redundant text. I had the feeling for some points such as purpose, description and vision you wrote the same things with other words so maybe try to avoid that.

Other than that I think your project has good intentions and that you thought it well through also with good mockups and that you already thought about ways to implement.

PS: I think you have chosen very interesting names, keep it going :D 

Yours truly
Vanessa (DHBW Community Dashboard)

**Reply by AlexMi-Ha** [reply]
Hi Vanessa,
thanks for your positive review!
Sadly we are programmers and not literary geniuses, so writing texts like these never comes natural to us. Maybe we will have to do some ROBBING (**R**edundancy **O**bliteration **B**ecause **B**etter **I**nstructions **N**eed **G**rammar) later.

Yours truly
Alex (CUMGroup - Project AMOGUS)

**Comment by DrKartoffel** [comment]
Hello Team AMOGUS,

your SRS and Use Case Diagram looks really good , great job on that! We have one question on the functionality though.
In your SRS you say "Players get a random Math problem", how do you generate those problems? Do have a database where you store pre defined Math problems or do you generate those "on the fly" with a an algorithm? Also why do you say you have almost(?) all mockups? Are there some mockups your team is divisive on whether to implement them or not, or are they maype difficult to implement?

-The HappyPlants Team

**Reply by AlexMi-Ha** [reply]
Hello there HappyPlants Team,
we plan to have a database for various problems ranging from easy to hard. But to keep it diverse we will also generate some problems on the fly according to a predefined pattern (like switching numbers etc). Some easy problems like mental arithmetic can also be generated completly independent.

We wrote "almost(?) all mockups" due to a late finish of one of the mockups for a core usecase. Additionally we of course will make more design mockups for less important pages to keep everything consistent to one style.

Yours truly
Alex (CUMGroup - Project AMOGUS)
